geeksay.js

```diff
@@ -155,15 +155,25 @@ function geeksay(text) {
 }
 
 function geeksayWord(text) {
+    const pattern = new RegExp(/[~`!#$%\^&*+=\-\[\]\\';,/{}|\\":<>\?]/);
+    let specialChar = '';
+
+    if (pattern.test(text[text.length - 1])) {
```

Comment:
Is this required?

Reply:
Do you mean declaring specialChar as an empty string?

Reply:
I meant the extra `if` check before the `while`.